base_app/admin.py

```diff
@@ -0,0 +1,173 @@
+import os
+from zipfile import ZipFile
+
+from django.contrib import admin
+from django.core.files.images import ImageFile
+from django import forms
+from django.shortcuts import render, redirect
+from django.urls import path
+
+from .models import Movie, ContactMessages, Hall, Schedule, Reservation, Cinema, Seat
+from cinema_project.utils import fetch_from_csv, send_email
+
+
+class CsvImport(forms.Form):
+    file_upload = forms.FileField()
+
+
+class MovieAdmin(admin.ModelAdmin):
+    list_display = ('name', 'poster_url', 'poster_image', 'description',
+                    'year', 'director', 'imdb_link', 'imdb_id', 'imdb_rating',
+                    'duration', 'trailer_id')
+    change_list_template = "admin/base_app/movie/change_list.html"
+
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path('import-csv/', self.import_csv),
+            path('import-zip/', self.import_zip)
+        ]
+        return my_urls + urls
```

Comment:
Nice! Did you do this just to keep the import methods inside the MovieAdmin class?

Reply:
This was a solution I found and adopted :) . I'm interested in learning a different approach if you have a suggestion! Why not?

Reply:
I thought this could also work with a standard view, mapped to an "admin-url/" URL but this looks cleaner.